datafusion/common/src/error.rs

```diff
@@ -73,6 +73,9 @@ pub enum DataFusionError {
     /// This error happens whenever a plan is not valid. Examples include
     /// impossible casts.
     Plan(String),
+    /// This error happens when an invalid or unsupported option is passed
+    /// in a SQL statement
+    InvalidOption(String),
```

Comment:
If this causes any issues with the ongoing work around DataFusionError, I'm happy to roll it back. I created this because I didn't feel any existing error type seemed to make sense for an invalid option.